build/xwalk.gni

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This file contains Crosswalk-feature-related build flags. Users should
+# import it into their build folder by run:
+#   echo 'import("//xwalk/build/xwalk.gni")' > out/your_folder/args.gn'
+# Then run:
+#   gn gen out/your_folder
+
+# Set it false when building content_shell.
+declare_args() {
```

Comment:
@rakuco   I submitted a PR based on the 3rd option.  The gyp file is also changed to avoid break our regular build. Please help to have a look.
I think I can continue the GN work with the approach of "3rd option" plus "root_extra_deps".  Thanks so much for your help!